src/components/Documentation/Markdown/index.tsx

```diff
@@ -30,17 +36,31 @@ const isInsideCodeBlock = (node: Element): boolean => {
 }
 
 const Details: React.FC<{
-  children: Array<{ props: { children: Array<string> } } | string>
+  children: Array<{ props: { children: ReactNode } } | string>
 }> = ({ children }) => {
   const filteredChildren = children.filter(child => child !== '\n')
 
   if (!filteredChildren.length) return null
   if (typeof filteredChildren[0] === 'string') return null
 
-  const text = filteredChildren[0].props.children[0]
+  const headingChildren: ReactNode[] = filteredChildren[0].props
+    .children as ReactNode[]
+
+  // Remove header auto-link if present
+  const finalHeadingChild = headingChildren[headingChildren.length - 1] as {
```

Comment:
not sure why do we filter only the last node - could you clarify how does it work please?

Reply:
Sure! It's kind of weird and a workaround to adapt this fix to our ecosystem.

Basically, this bit removes the link icon that `gatsby-remark-autolink-headers` adds on the headings that eventually become the trigger for a `details` box. Since `details` is a component and the autolinker is a Remark plugin, there's no way to prevent the headers from getting an auto-link from within our codebase.

To prevent our `details` component from having to do this, we'd need to change how the header auto-links work by forking the auto-link plugin or rebuilding the functionality ourselves, which is totally doable but obviously more intensive than with this quick fix.

The reason this wasn't an issue before was that the previous behavior sliced off all but the first element, which is very similar to what this is doing but with the caveat of markup causing the break that this fix is for.

Now that I think about it though, if all `details` blocks are started with headers that are always auto-linked, this could just be a simple `slice` that removes the last item each time and skip the ternary.

Reply:
> Now that I think about it though, if all details blocks are started with headers that are always auto-linked, this could just be a simple slice that removes the last item each time and skip the ternary.

 May be there should be some check in place that check `filteredChildren[0]` type properly (not only `=== string`)? Otherwise looks like the main method expects it to be the header.

Reply:
Yeah, `filteredChildren[0]` is expected to be the MD heading in the `details` that becomes the `Collapsable` trigger. I can make a change to more specifically make the details box only when the first element is a heading. I'm not sure if `details` is used in any other way, but if not we could either throw or provide a default `trigger` text for `details` that don't lead with a heading.

Reply:
I took your advice and now we check that the first child is a heading, regardless of its level. Otherwise, the component throws and is caught at build-time with an error message informing the user a heading must be used.

While the throw behavior is only for source readability, I think we want to keep `details` blocks uniform in the sense that they're always started with a heading instead of accepting any first as `trigger` and letting them become inconsistent.